naucse/routes.py

```diff
@@ -117,13 +117,35 @@ def course(course):
     def lesson_url(lesson, *args, **kwargs):
         return url_for('course_page', course=course, lesson=lesson, *args, **kwargs)
 
+    recent_runs = []
+    if not course.start_date:
```

Comment:
I'd make this a function/method/whatnot as well (see the next comment that I made earlier). It makes the route (it's code) too long.